### DISCUSSION TITLE
Need help in mantine component props extend

### DISCUSSION THREAD

**Discussion by Nikhilchandra-Rathod** [mantinedev/mantine#2962] [Q&A]
I am creating a wrapper component for the Mantine component as follows.

import { Button, ButtonProps } from '@mantine/core';

/* eslint-disable-next-line */
export type UButtonProps = {} & ButtonProps;

export function UButton(props:  UButtonProps) {
  return <Button {...props} />;
}

export default UButton;

and After that I am using it in another page as follows.
<UButton onClick={() => console.log('click!!!')}>Click Me</UButton>

but it showed me the following typescript error.
"message": "Type '{ children: string; onClick: () => void; }' is not assignable to type 'IntrinsicAttributes & { css?: Interpolation<Theme>; } & ButtonProps'.\n  Property 'onClick' does not exist on type 'IntrinsicAttributes & { css?: Interpolation<Theme>; } & ButtonProps'.",

Can you please suggest me, How can I extend mantine component props ?
I need all the props from Mantine Component and also extend my own props here.
export type UButtonProps = {} & ButtonProps;

**Comment by rtivital** [comment]
`interface MyButtonProps extends ButtonProps, React.ComponentPropsWithoutRef<'button'> {}`

**Reply by kaisiant** [reply]
<img width="254" alt="image" src="https://user-images.githubusercontent.com/26655111/202839451-162037e4-c568-41c9-83d9-8ac1efa0ed73.png">

Need some help. Having a hard time creating reusable components for the library. With ChakraUI, I just need to import the ButtonProps.  I guess is because of the flexibility and complexity of mantine UI is offering compare to Chakra.

**Reply by rtivital** [reply]
https://mantine.dev/guides/polymorphic/#create-your-own-polymorphic-components

**Reply by Nikhilchandra-Rathod** [reply]
> <img alt="image" width="254" src="https://user-images.githubusercontent.com/26655111/202839451-162037e4-c568-41c9-83d9-8ac1efa0ed73.png">
> 
> Need some help. Having a hard time creating reusable components for the library. With ChakraUI, I just need to import the ButtonProps. I guess is because of the flexibility and complexity of mantine UI is offering compare to Chakra.

```
export type YourButtonProps<T extends React.ElementType> = {
  component?: T;
} & ButtonProps &
  React.ComponentPropsWithoutRef<T>;


export function YourButton<T extends React.ElementType>(
  props: YourButtonProps<T>
) {
  return <Button {...props} />;
}
```

**Reply by kaisiant** [reply]
> > <img alt="image" width="254" src="https://user-images.githubusercontent.com/26655111/202839451-162037e4-c568-41c9-83d9-8ac1efa0ed73.png">
> > Need some help. Having a hard time creating reusable components for the library. With ChakraUI, I just need to import the ButtonProps. I guess is because of the flexibility and complexity of mantine UI is offering compare to Chakra.
> 
> ```
> export type YourButtonProps<T extends React.ElementType> = {
>   component?: T;
> } & ButtonProps &
>   React.ComponentPropsWithoutRef<T>;
> 
> 
> export function YourButton<T extends React.ElementType>(
>   props: YourButtonProps<T>
> ) {
>   return <Button {...props} />;
> }
> ```

freaking awesome
